cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -842,6 +877,47 @@ TEST_F(TestRecordBatchSortIndices, NaNAndNull) {
   AssertSortIndices(batch, options, "[7, 1, 2, 6, 5, 4, 0, 3]");
 }
 
+TEST_F(TestRecordBatchSortIndices, Boolean) {
```

Comment:
I don't think it's worth testing the non-null case separately. This will make less test code to maintain.